src/main/java/org/generationcp/middleware/dao/WorkbenchUserDAO.java

```diff
@@ -424,22 +421,7 @@ public List<WorkbenchUser> getUsersByPersonFirstNameOrLastNameContains(final Str
 
 	public long countSearchUsers(final UserSearchRequest userSearchRequest) {
 		try {
-			final Criteria criteria = this.getSession().createCriteria(WorkbenchUser.class);
-
-			criteria.createAlias("person", "person", CriteriaSpecification.LEFT_JOIN);
-			criteria.createAlias("person.crops", "crops", CriteriaSpecification.LEFT_JOIN);
-
-			if (userSearchRequest.getStatus() != null) {
-				criteria.add(Restrictions.eq("status", userSearchRequest.getStatus()));
-
-			}
-
-			if (userSearchRequest.getRoleId() != null) {
-				criteria.createAlias("roles", "roles");
-				criteria.add(Restrictions.eq("roles.id", userSearchRequest.getRoleId()));
-			}
-
-			criteria.setResultTransformer(CriteriaSpecification.DISTINCT_ROOT_ENTITY);
+			final Criteria criteria = this.createUserCriteria(userSearchRequest);
 			return criteria.list().size();
```

Comment:
Use Projections.rowCount to count the result:

criteria.setProjection(Projections.rowCount());
Integet count = (Integer)criteria.uniqueResult()